### PR TITLE
DOC: update the Period.dayofyear docstring

### PR DESCRIPTION
**Pull request by techytushar**
Signed-off-by: Tushar Mittal <chiragmittal.mittal@gmail.com>

Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

- [x] PR title is "DOC: update the <your-function-or-method> docstring"
- [x] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [x] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [x] It has been proofread on language by another sprint participant

Please include the output of the validation script below between the "```" ticks:

```
################################################################################
##################### Docstring (pandas.Period.dayofyear)  #####################
################################################################################

Return the day of the year.

This attribute returns the day of the year on which the particular
date occurs. The return value ranges between 1 to 365 for regular
years and 1 to 366 for leap years.

Returns
-------
int
    The day of year.

See Also
--------
Period.dayofweek : Return the day of week.
Period.daysinmonth : Return the days in that month.
PeriodIndex.dayofyear : Return the day of year of all indexes. 

Examples
--------
>>> period = pd.Period("2015-10-23", freq='H')
>>> period.dayofyear
296
>>> period = pd.Period("2012-12-31", freq='D')
>>> period.dayofyear
366
>>> period = pd.Period("2013-01-01", freq='D')
>>> period.dayofyear
1

################################################################################
################################## Validation ##################################
################################################################################

Docstring for "pandas.Period.dayofyear" correct. :)
```

If the validation script still gives errors, but you think there is a good reason
to deviate in this case (and there are certainly such cases), please state this
explicitly.

### CODE REVIEW

**Comment by techytushar** [comment]
@jreback please review :slightly_smiling_face:

File: pandas/_libs/tslibs/period.pyx [hunk 1]
```diff
@@ -1255,6 +1255,33 @@ cdef class _Period(object):
...
+        """
+        Return the day of the year.
+
+        This attribute returns the day of the year on which the particular
+        date occurs.
+
+        Returns
+        -------
+        int
+        The day of year.
```
Comment (TomAugspurger) [inline pandas/_libs/tslibs/period.pyx]: This should be indented 4 space.

File: pandas/_libs/tslibs/period.pyx [hunk 2]
```diff
@@ -1255,6 +1255,33 @@ cdef class _Period(object):
...
+
+        See Also
+        --------
+        Period.dayofweek : Return the day of week.
+        Period.daysinmonth : Return the days in that month.
+
```
Comment (TomAugspurger) [inline pandas/_libs/tslibs/period.pyx]: Add PeriodIndex.dayofyear

File: pandas/_libs/tslibs/period.pyx [hunk 3]
```diff
@@ -1255,6 +1255,33 @@ cdef class _Period(object):
 
     @property
     def dayofyear(self):
+        """
+        Return the day of the year.
```
Comment (TomAugspurger) [inline pandas/_libs/tslibs/period.pyx]: Periods represent time spans so it's good to note that this is the start of the period. So maybe

    Return the day of the year the of the Period's start.

That's a bit wordy though. Feel free to reword if you have a better phrasing.

File: pandas/_libs/tslibs/period.pyx (hunk 2 above)
Comment (jorisvandenbossche) [reply pandas/_libs/tslibs/period.pyx]: Do you mean 'weekofyear' ? (dayofyear is the current docstring)

File: pandas/_libs/tslibs/period.pyx (hunk 3 above)
Comment (techytushar) [reply pandas/_libs/tslibs/period.pyx]: sorry, but I didn't understand the meaning of the sentence `Return the day of the year the of the Period's start.`, specifically of the `period's start`

**Comment by techytushar** [comment]
made the required changes, please review @jorisvandenbossche :slightly_smiling_face:

File: pandas/_libs/tslibs/period.pyx (hunk 3 above)
Comment (TomAugspurger) [reply pandas/_libs/tslibs/period.pyx]: Period is a span, so saying "Day of the year" is a bit ambiguous. Is it the start, end, somewhere in between?

If you have a period like `Period('2017-01-01', freq='A')

```python
In [2]: pd.Period('2017-01-01', freq='A')
Out[2]: Period('2017', 'A-DEC')

In [3]: pd.Period('2017-01-01', freq='A').day
Out[3]: 31
```

I was apparently wrong about it being the start. I guess it's related to `freq`

Comment (jorisvandenbossche) [reply pandas/_libs/tslibs/period.pyx]: This comment will be relevant to many of the Period attributes like day, dayofyear, .. (also the ones I already merged ..). 
So it would be good to be consistent in it.

The annual frequency is by default a "year-end" frequency, so it's indeed freq related.

Comment (dukebody) [reply pandas/_libs/tslibs/period.pyx]: I'm sorry, I don't follow... Can you show two cases creating a period starting on the same day but with different frequencies, and how this affect `.day`, `.dayofweek`, `.dayofyear`, etc.?

Comment (jorisvandenbossche) [reply pandas/_libs/tslibs/period.pyx]: @dukebody small example:

```
In [30]: p1 = pd.Period('2017-01-01', freq='D')

In [31]: p2 = pd.Period('2017-01-01', freq='M')

In [32]: p1
Out[32]: Period('2017-01-01', 'D')

In [33]: p2
Out[33]: Period('2017-01', 'M')

In [34]: p1.start_time
Out[34]: Timestamp('2017-01-01 00:00:00')

In [35]: p2.start_time
Out[35]: Timestamp('2017-01-01 00:00:00')

In [36]: p1.dayofyear
Out[36]: 1

In [37]: p2.dayofyear
Out[37]: 31
```

I am not really familiar with this frequency business, but you have certain frequences which are "anchored" at the beginning of the period, and others at the end of the period. But I don't directly see a way how to inspect this.

**Review by jorisvandenbossche — APPROVED** [review]
Apart from the discussion about start/end in the summary line, looks good to me!

File: pandas/_libs/tslibs/period.pyx (hunk 3 above)
Comment (dukebody) [reply pandas/_libs/tslibs/period.pyx]: That's quite confusing. Should we open a ticket to make `.dayofyear` etc. always refer to the start time of the period? Not to the "anchored" time. Otherwise these functions are very unpredictable IMO and therefore don't add a lot of value.

Comment (jorisvandenbossche) [reply pandas/_libs/tslibs/period.pyx]: Yes, I am writing one currently ;) 
I suppose those attributes will mainly be useful for frequencies <= 1D

**Comment by jorisvandenbossche** [comment]
> Apart from the discussion about start/end in the summary line

Let's merge this, and will open a new issue to further discuss the confusion around start/end of the period. We can later fix the wording of the summary line.

File: pandas/_libs/tslibs/period.pyx (hunk 3 above)
Comment (dukebody) [reply pandas/_libs/tslibs/period.pyx]: I'm OK with merging this PR without referring to the start or end date of the period, and then work on code to make this predictable and change all docstrings to refer to the period start date.

**Comment by jorisvandenbossche** [comment]
@techytushar Thanks a lot!

**Comment by jorisvandenbossche** [comment]
I opened an issue here: https://github.com/pandas-dev/pandas/issues/20324
